grails-app/controllers/com/unifina/controller/api/KeyApiController.groovy

```diff
@@ -174,19 +186,19 @@ class KeyApiController {
 					Permission.Operation operation = Permission.Operation.fromString(permission)
 					SecUser user = request.apiUser
 					boolean logIfDenied = false
-					if (operation == Permission.Operation.READ) {
-						if (!hasPermission(Permission.Operation.READ, k.getPermissions())) {
-							permissionService.grant(user, res, k, Permission.Operation.READ, logIfDenied)
+					if (operation == Permission.Operation.STREAM_GET) {
+						if (!hasPermission(Permission.Operation.STREAM_GET, k.getPermissions())) {
+							permissionService.grant(user, res, k, Permission.Operation.STREAM_GET, logIfDenied)
 						}
-						if (hasPermission(Permission.Operation.WRITE, k.getPermissions())) {
-							permissionService.revoke(user, res, k, Permission.Operation.WRITE, logIfDenied)
+						if (hasPermission(Permission.Operation.STREAM_EDIT, k.getPermissions())) {
+							permissionService.revoke(user, res, k, Permission.Operation.STREAM_EDIT, logIfDenied)
 						}
-					} else if (operation == Permission.Operation.WRITE) {
-						if (!hasPermission(Permission.Operation.READ, k.getPermissions())) {
-							permissionService.grant(user, res, k, Permission.Operation.READ, logIfDenied)
+					} else if (operation == Permission.Operation.STREAM_EDIT) {
```

Comment:
`if (operation == STREAM_PUBLISH)` should grant (if not already granted) `STREAM_GET`, `STREAM_SUBSCRIBE`, and `STREAM_PUBLISH`